packages/core/src/lib/chainStreamManager/ChainStreamConsumer/consumeStream.ts

```diff
@@ -29,20 +29,79 @@ interface ConsumeStreamParams {
 }
 
 interface ConsumeStreamResult {
-  error?: ChainError<RunErrorCodes.AIRunError>
+  error?:
+    | ChainError<RunErrorCodes.AIRunError>
+    | ChainError<RunErrorCodes.RateLimit>
   finishReason: FinishReason
 }
 
+export async function consumeStream({
+  controller,
+  result,
+}: ConsumeStreamParams): Promise<ConsumeStreamResult> {
+  let error:
+    | ChainError<RunErrorCodes.AIRunError>
+    | ChainError<RunErrorCodes.RateLimit>
+    | undefined
+  let finishReason: FinishReason = 'stop'
+
+  for await (const chunk of streamToGenerator<StreamChunk>(result.fullStream)) {
+    if (chunk.type === 'error') {
+      finishReason = 'error'
+      error = createAIError(
+        getErrorMessage({
+          error: chunk.error,
+          providerName: result.providerName,
+        }),
+        getErrorCode(chunk.error),
+      )
+
+      break
+    }
+
+    if (chunk.type === 'finish') {
+      finishReason = chunk.finishReason
+
+      if (chunk.finishReason === 'error') {
+        error = createAIError('AI run finished with error')
+        break
+      }
+    }
+
+    enqueueChainEvent(controller, {
+      event: StreamEventTypes.Provider,
+      data: chunk,
+    })
+  }
+
+  return { error, finishReason }
+}
+
+function getErrorCode(error: unknown) {
+  if (error instanceof APICallError) {
+    if (error.statusCode === 420) {
```

Comment:
LOL. 
<img width="979" alt="image" src="https://github.com/user-attachments/assets/cd66fb49-70b3-4a38-978d-9ae55d4dfcc1" />


Is this what vercel SDK returns?